### PR TITLE
tests: give http_proxies enough time to bootstrap and use randomized port

### PR DESCRIPTION
**Pull request by YuJuncen**
### What problem does this PR solve? 
In bad network environment, `http_proxies` would fail consistently.

### What is changed and how it works?
Give it sometime to bootstrap, use randomized port, or, skip this test.

### Check List 

Tests 

 - Integration test

### Release note

- No release note

### CODE REVIEW

File: tests/http_proxies/run.sh [hunk 1]
```diff
@@ -49,13 +50,18 @@ function prepare() {
     start_tidb_cluster --workdir $WORK_DIR
 
     start_proxy
+    # TODO: make sure the proxy is started.
+    sleep 20
```
Comment (amyangfei) [inline tests/http_proxies/run.sh:54]: Could you add some retry mechanism for proxy checking instead of sleep 20s

Comment (YuJuncen) [reply tests/http_proxies/run.sh]: done

**Review by amyangfei — APPROVED** [review]

**Comment by YuJuncen** [comment]
FAIL: sorter_test.go:318: sorterSuite.TestSorterIOError

/run-integartion-tests

**Comment by YuJuncen** [comment]
...Again? 🤔

/run-integartion-tests

**Comment by amyangfei** [comment]
> ...Again? 🤔
> 
> /run-integartion-tests

typo
s/integartion/integration

**Comment by YuJuncen** [comment]
`etcdSuite.TestStreamRecvWithErrorAndResolvedGoBack`

/run-leak-test

**Comment by YuJuncen** [comment]
`sorterSuite.TestSorterIOError`

/run-unit-tests

**Comment by YuJuncen** [comment]
`Failed connect to 127.0.0.1:24927; Connection refused` 🤔

/run-integration-tests

**Pull request merged**